test/test_openapi.py

```diff
@@ -120,6 +121,67 @@ def route2():
                 pass
         self.assertEqual(str(cm.exception), 'Only specify path_spec once per route path')
 
+    def test_shared_path_spec(self):
+        app = AzulChaliceApp('testing', spec={'foo': 'bar'})
+        swagger_test_path_spec = {
+            'parameters': [
+                params.query('foo', schema.optional({'type': 'string'})),
+            ]
+        }
+
+        @app.route('/swagger-test-1', methods=['GET'], cors=True, path_spec=swagger_test_path_spec, method_spec={
+            'summary': 'Swagger test 1',
+            'description': 'Swagger test 1',
+            'responses': {'201': {'description': 'OK'}}
+        })
+        def swagger_test_1():
+            pass
+
+        @app.route('/swagger-test-2', methods=['GET'], cors=True, path_spec=swagger_test_path_spec, method_spec={
+            'summary': 'Swagger test 2',
+            'description': 'Swagger test 2',
+            'responses': {'202': {'description': 'OK'}}
+        })
+        def swagger_test_2():
+            pass
+
+        expected_spec = {
```

Comment:
If possible the assertion should not cover the other parts of the output, only that the `foo` path parameter is the same in the two routes.

Reply:
We want the assertion to cover all parts of the output. The original issue was that the method spec from the later route overwrote the method spec of the first one, therefore we need to assert the method spec as well.

Reply:
I added a docstring that hopefully makes that more clear.

Reply:
It does.

Reply:
I take that back. If the goal of the test is to ensure that method specs are not overwritten—which would cause two things to be the *same* that shouldn't be—then the assertion should use assertNotEqual on the method specs.

Reply:
Good point.